sos/plugins/juju.py

```diff
@@ -9,113 +9,201 @@
 # See the LICENSE file in the source distribution for further information.
 
 import os
+import re
+import glob
 from sos.plugins import Plugin, UbuntuPlugin
 from json import loads as json_loads
 
 
-def ensure_service_is_running(service):
-    def wrapper(callback):
-        def wrapped_f(self, *args, **kwargs):
-            try:
-                result = self.call_ext_prog("service {0} stop".format(service))
-                if result["status"] != 0:
-                    raise Exception("Cannot stop {0} service".format(service))
-                callback(self, *args, **kwargs)
-            except Exception as ex:
-                self._log_error("Cannot stop {0}, exception: {1}".format(
-                    service,
-                    ex.message))
-            finally:
-                self.call_ext_prog("service {0} start".format(service))
-        return wrapped_f
-    return wrapper
-
-
 class Juju(Plugin, UbuntuPlugin):
     """ Juju orchestration tool
     """
 
-    plugin_name = 'juju'
-    profiles = ('virt', 'sysmgmt')
-    files = ('/usr/bin/juju', '/usr/bin/juju-run')
+    plugin_name = "juju"
+    profiles = ("virt", "sysmgmt")
+    files = ("/snap/bin/juju", "/usr/bin/juju")
 
     option_list = [
-        ('export-mongodb',
-         'Export mongodb collections as json files', '', False),
-        ('generate-bundle',
-         """Generate a YAML bundle of the current environment
-         (requires juju-deployerizer)""", '', False),
```

Comment:
Note that if you are deleting plugin options, then any automation that tries to use them will break with the updated plugin. We tend to avoid this unless absolutely necessary for this reason.